lib/wip/product_mapping_exporter.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'sequel'
+
+module Magento
+  class ProductMappingExporter
+    ADAPTERS = {
+      'mysql': :mysql2,
+      'postgres': :postgres,
+    }
+
+    BATCH_SIZE = 1000
+
+    def initialize(
+      database_type: :mysql,
+      database_name:,
+      host: '127.0.0.1',
+      port: 3306, username:,
+      password: '',
+      filename: 'magento_product_mappings.csv'
+    )
+      raise 'Invalid adapter' unless ADAPTERS.keys.include? database_type
+
+      @adapter = ADAPTERS[database_type]
+      @host = host
+      @port = port
+      @username = username
+      @password = password
+      @filename = filename
+      @database_name = database_name
+
+      write_headers
+    end
+
+    def extract_mappings
+      db = Sequel.connect(
+        adapter: adapter, user: username, password: password, host: host, port: port, database: database_name
+      )
+
+      mappings = db.from(:catalog_product_relation)
+      ordered_mappings = mappings.order(:parent_id)
+      current_id = ordered_mappings.first[:parent_id]
+      max_id = ordered_mappings.last[:parent_id]
+
+      while current_id <= max_id do
+        puts "doing batch #{current_id}"
```

Comment:
May want to use `STDERR` here, esp. if we move to converting this to a thor subcommand and the output could go to `STDOUT`, like the other subcommands.

Reply:
woops, this was added just for debugging purposes, I forgot to remove it.